mitiq/zne/inference.py

```diff
@@ -1363,7 +1363,15 @@ def extrapolate(  # type: ignore
 
         # Deduce "sign" parameter of the exponential ansatz
         linear_params, _ = mitiq_polyfit(scale_factors, exp_values, deg=1)
-        sign = np.sign(-linear_params[0])
+
+        if (
+            asymptote is not None
+            and np.sign(-linear_params[0] * (asymptote - linear_params[1]))
+            == 1
+        ):
+            sign = np.sign(linear_params[0])
```

Comment:
This part of the code seems equivalent to the following simplified code:

```suggestion
        if asymptote:
            sign = np.sign(asymptote - linear_params[1])
```

But I may be wrong. Please double-check.

Reply:
Almost! Thanks for suggesting I simplify this :)